src/main/java/bio/terra/service/snapshot/flight/create/SnapshotAuthzBqJobUserStep.java

```diff
@@ -54,10 +45,11 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
         new ArrayList<>(List.of(policyMap.get(IamRole.STEWARD), policyMap.get(IamRole.READER)));
 
     if (sourceDataset.isInheritSteward()) {
-      var datasetPolicyMap =
-          sam.retrievePolicyEmails(request, IamResourceType.DATASET, sourceDataset.getId());
+      Map<IamRole, String> sourceDatasetPolicyMap =
+          workingMap.get(
+              SnapshotWorkingMapKeys.SOURCE_DATASET_POLICY_MAP, new TypeReference<>() {});
       // Allow the custodian to make queries in this project.
-      policyEmails.add(datasetPolicyMap.get(IamRole.CUSTODIAN));
+      policyEmails.add(sourceDatasetPolicyMap.get(IamRole.CUSTODIAN));
```

Comment:
If the only element of the map that's needed is `CUSTODIAN`, why not only add that entry? That's what `EnableInheritStewardFlight` does

Reply:
I liked following the pattern of putting the whole policy map into the flight map like we're doing for the snapshot policy map.